File: csharp/Structural/07-Bridge/Notification/HighNotification.cs
using Bridge.Channel;

namespace Bridge.Notification;

public class HighNotification : JobNotification
{
    public HighNotification(INotificationChannel channel) : base(channel) { }

    public override void Notify(string jobId, string recipient, string message)
    {
        Channel.Send(recipient, $"[HIGH PRIORITY] Job #{jobId}", message);
    }
}

File: typescript/Structural/07-Bridge/Notification/HighNotification.ts
import { INotificationChannel } from "../Channel/INotificationChannel";
import { JobNotification } from "./JobNotification";

export class HighNotification extends JobNotification {
  constructor(channel: INotificationChannel) {
    super(channel);
  }

  notify(jobId: string, recipient: string, message: string): void {
    this.channel.send(recipient, `[HIGH PRIORITY] Job #${jobId}`, message);
  }
}
